File: src/main/java/ru/stnk/RestTestAPI/repository/InstrumentEntityRepository.java
package ru.stnk.RestTestAPI.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.stnk.RestTestAPI.entity.coins.InstrumentEntity;

@Repository
public interface InstrumentEntityRepository extends JpaRepository<InstrumentEntity, Long> {
    InstrumentEntity findBySymbolName(String symbolName);
}

File: src/main/kotlin/ru/stnk/resttestapi/repository/InstrumentEntityRepository.kt
package ru.stnk.resttestapi.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.stnk.resttestapi.entity.coins.InstrumentEntity

@Repository
interface InstrumentEntityRepository : JpaRepository<InstrumentEntity, Long> {
    // Указанание на то что эта функция может вернуть null важно
    fun findBySymbolName(symbolName: String): InstrumentEntity?
}
